### PR TITLE
Add delete indexed document

### PR DESCRIPTION
**Pull request by nicolasThal**

### CODE REVIEW

File: Elastica/EventSubscriber/UpdateContentIndexedSubscriber.php [hunk 1]
```diff
@@ -33,14 +34,12 @@ public function updateIndexedContent(ContentEvent $event)
...
 
-        if (!$content->getStatus()->isPublished()) {
-            return;
-        }
-
         $lastPublishedContent = $this->contentRepository->findLastPublishedVersion($content->getContentId(), $content->getLanguage());
 
-        if ($content->getVersion() >= $lastPublishedContent->getVersion()) {
-            $this->contentIndexor->index($content);
+        if ($lastPublishedContent instanceof ContentInterface) {
```
Comment (alavieille) [inline Elastica/EventSubscriber/UpdateContentIndexedSubscriber.php:39]: why update the index if the content isn't published ?

Comment (nicolasThal) [inline Elastica/EventSubscriber/UpdateContentIndexedSubscriber.php:39]: If you unpublish a content, you will have to update the index to the last published content or you will have to purge the index from the content

Comment (alavieille) [inline Elastica/EventSubscriber/UpdateContentIndexedSubscriber.php:39]: Yes but, if you change a content in other status that published like pending, the you index the lastPublishedContent whereas already indexed

Comment (ngilain) [inline Elastica/EventSubscriber/UpdateContentIndexedSubscriber.php:39]: I think too there could be an optimization here. There is three cases :
1/ the event gives a published content => index
2/ the event gives a content changing from status publish to something else => delete
3/ all other cases => do nothing

Comment (nicolasThal) [inline Elastica/EventSubscriber/UpdateContentIndexedSubscriber.php:39]: @itkg-ngilain, I think that the cases you listed are not full : 
- what do I do if the Content before the content included in the event is still publish ?
- what do I do if the content in the event is published but is not the last published version of the content ?

**Comment by ngilain** [comment]
:+1:

**Pull request merged**